File: questao2.py
#!/usr/bin/python 

def expoente(e):
	result = 1
	for i in range(1,e+1):
		result *= e
	return result
	
vetor = []
resultado = 0;
n = input('Digite um valor para N: ')
if(n%2==0):
	n = n/2
else:
	n = (n+1)/2

for i in range (1,n+1):
	aux1 = float(expoente((2*i)-1))
	aux2 = float(expoente(2*i))
	vetor.append(float(aux1/aux2))
	if(i-1!=0):
		if((i-1)%2==0):
			aux1 = vetor[i-2]
			aux2 = vetor[i-1]
			vetor[i-2] = aux1*aux2
			vetor[i-1] = 0

for i in range(1,n+1):
	resultado += vetor[i-1]

print resultado

File: questao2.cpp
#include <iostream>
#include <cstdlib>
using namespace std;

int Expoente(int e){
	int i;
	int result=1;
	for(i=1;i<=e;i++){
		result*=e;
	}
	return result;
}

int main(){
	int n,i,nvezes;
	double *v,aux1,aux2,resultado;
	cin >> n;
	v = (double*)malloc(n*sizeof(double));
	if(n%2==0){
		n = n/2;
	}else{
		n = (n+1)/2;
	}
	for(i=1;i<=n;i++){
		aux1 = Expoente((2*i)-1);
		aux2 = Expoente(2*i);
		v[i-1] = aux1/aux2;
		
		if(i-1!=0){	
			if((i-1)%2==0){
				aux1 = v[i-2];
				aux2 = v[i-1];
				v[i-2] = aux1*aux2;
				v[i-1] = 0;
			}
		}
	}
	for(i=1;i<=n;i++){
		resultado += v[i-1];
	}
	cout <<"\nResultado: "<<resultado<<"\n";
	return 0;
}
